### PR TITLE
[Bugfix] try to fold constant during castOutputColumnsTypeToTargetColumns in InsertPlanner

### LINKED ISSUE #9881: [Bug] insert into select null not supported in 2.3.0 ?
<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
```
mysql> show create table t2;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                                                                                                                                               |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t2    | CREATE TABLE `t2` (
  `c1` int(11) NULL COMMENT "",
  `c2` int(11) REPLACE_IF_NOT_NULL NULL COMMENT "",
  `c3` ARRAY<int(11)> REPLACE_IF_NOT_NULL NULL COMMENT ""
) ENGINE=OLAP 
AGGREGATE KEY(`c1`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`c1`) BUCKETS 1 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
); |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

mysql> insert into t2 select c1+2, c2, null from t2;
ERROR 1064 (HY000): Vectorized engine does not support the operator, node_type: 5

### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9881

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|5|5|100.00%|[]|

**Review by Seaven — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java [hunk 1]
```diff
@@ -308,13 +312,15 @@ private OptExprBuilder castOutputColumnsTypeToTargetColumns(ColumnRefFactory col
                                                                 OptExprBuilder root) {
         List<Column> fullSchema = insertStatement.getTargetTable().getFullSchema();
         Map<ColumnRefOperator, ScalarOperator> columnRefMap = new HashMap<>();
-
+        ScalarOperatorRewriter rewriter = new ScalarOperatorRewriter();
+        List<ScalarOperatorRewriteRule> rewriteRules = Arrays.asList(new FoldConstantsRule());
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java:316]: Why Optimizer.optimize cannot handle this? still add a rule here?

**Review by murphyatwork — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java (hunk 1 above)
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java:316]: > Why Optimizer.optimize cannot handle this? still add a rule here?

There seems to be no special rule processing for this scenario, it is a very local problem, so I just did rewrite here……do you have a better suggestion? @Seaven

**Comment by silverbullet233** [comment]
https://github.com/Mergifyio backport branch-2.3

**Comment by silverbullet233** [comment]
https://github.com/Mergifyio backport branch-2.4

File: fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java:316]: Optimizer can only handle query stmt. The target table of insert stmt is invisible for optimizer, so only handle it in here

**Pull request merged**